src/backend/aspen/app/views/sample.py

```diff
@@ -531,31 +534,52 @@ def update_sample_public_ids():
             f"Private Identifiers {missing_private_identifiers} not found in DB",
         )
 
-    # check that public_identifiers don't already exist
-    existing_public_ids: list[str] = api_utils.get_existing_public_ids(
-        request_public_ids, g.db_session, group_id=group_id
+    samples_to_update = g.db_session.query(Sample).filter(
+        and_(
+            Sample.submitting_group_id == group_id,
+            Sample.private_identifier.in_(private_to_public.keys()),
+        )
     )
-    if existing_public_ids:
-        raise ex.BadRequestException(
-            f"Public Identifiers {existing_public_ids} are already in the database",
+
+    if public_ids_are_gisaid_isl:
+        samples_to_update = samples_to_update.options(
+            joinedload(Sample.uploaded_pathogen_genome),
         )
+        for s in samples_to_update:
+            isl_number = private_to_public[s.private_identifier]
+            accessions = s.uploaded_pathogen_genome.accessions()
+
+            # if the accessions length == 1 that means that currently no isl number exists on the uploaded_pathogen_genome
+            if not len(accessions) == 1:
+                # first accession public_identifier is 'gisaid_public_identifier', the second accession returned is the one that we want to update
+                accessions[1].public_identifier = isl_number
```

Comment:
I'm not quite sure why this behavior is like this.. @jgadling or @lvreynoso let me know if you know of a better way to do this

Reply:
We support (at least) GISAID and GENBANK accessions -- you're going to have to iterate over the accessions associated with the sample and find the one with the right type.

Reply:
That's the sample string used in the test's factory function for creating an `UploadedPathogenGenome`, that should be treated as an actual accession when you're testing things.  I think you would need something like 
```python
for accession in accessions:
    if isinstance(accession, GisaidAccession):
        ...
```
to find what you're looking for.

Reply:
Going by the currently existing database model this isn't strictly kosher, but since we're going to change how we handle GisaidAccessions very soon anyway I think it would be fine

Reply:
oh that was the test factory function accession.. that makes more sense. They were both of type `GisaidAccession` and that's why i didn't initially use this check.

Reply:
yeah this is a very temporary solution

Reply:
yeah they're both GisaidAccession objects, at least in the testing DB
<img width="768" alt="Screen Shot 2021-11-10 at 9 26 18 AM" src="https://user-images.githubusercontent.com/13052132/141162708-b6cb66c1-2d50-47f1-b3ce-633c0dcb6280.png">

Reply:
I think (??) we only expect one gisaid accession associated with a sample. I'm not sure what's up with the testing db. Would it make sense to update both gisaid accession identifiers?

Either way, we *do* want to make sure we're only updating gisaid accessions, so blindly using the second accession isn't going to be safe.

Reply:
i can do the check and then update all of the associate gisaid accessions with that sample. I didn't _want_ to blindly use the second accession, it just seemed like that was the only way to grab the correct accession to update since they were both of the same type. Pushing a change now

Reply:
I think you're right that there should only be one gisaid accession associated with each sample, i think the reason for multiple accessions on a sample could be for the use case of a sample being uploaded to both gisaid and genbank? idk